packages/application/src/index.ts

```diff
@@ -525,6 +533,31 @@ export class Application<T extends Widget = Widget> {
     return this._delegate.promise;
   }
 
+  /**
+   * Collect the deferred plugins.
+   *
+   * @returns The list of all the deferred plugins.
+   */
+  deferredPlugins(): string[] {
+    return Array.from(this._plugins.keys()).filter(
+      id => this._plugins.get(id)!.deferred
+    );
```

Comment:
FYI you could use the map directly in this case to avoid the use of `!`:

```suggestion
    return Array.from(this._plugins).filter(
      ([id, plugin]) => plugin.deferred
    ).map(([id, plugin]) => id);
```